### ISSUE TITLE
Multiple cache-control headers are not handled

### ISSUE THREAD

**Issue by tjarvstrand** [ivoleitao/stash#14]
**Describe the bug**
While debugging some code I came across a situation where the server returned a response with multiple cache-control header values. This is valid according to RFC2616 but it will currently cause `stash_dio` to throw an exception (`'"cache-control" header has more than one value, please use Headers[name]'`)

**To Reproduce**
Make a request to an endpoint returning several cache-control header values

**Expected behavior**
The values are concatenated, interspersed with commas

**Version**
Flutter 2.2.0 • channel stable • https://github.com/flutter/flutter.git
Framework • revision b22742018b (5 weeks ago) • 2021-05-14 19:12:57 -0700
Engine • revision a9d88a4d18
Tools • Dart 2.13.0

**Additional context**
[Relevant section of RFC2616](https://datatracker.ietf.org/doc/html/rfc2616#section-14.9)
[Stack overflow question](https://stackoverflow.com/questions/4371328/are-duplicate-http-response-headers-acceptable)

**Comment by tjarvstrand** [comment]
This interceptor seems to function as a workaround:
```
static final _cacheControlWrapper = InterceptorsWrapper(
    onResponse: (response, handler) {
      final cacheControl = response.headers["cache-control"];
      if (cacheControl != null && cacheControl.length > 1) {
        response.headers.set("cache-control", cacheControl.join(","));
      }
      handler.next(response);
    },
  );
```

**Comment by ivoleitao** [comment]
Thanks for the report @tjarvstrand I will take a look at this during the weekend

**Comment by ivoleitao** [comment]
Fixed, it was easy to simulate although your complete description helped a lot. I've added a unit test for this and actually used a similar solution as the one you employed on the workaround as per RFC spec two separate cache-control headers are the same as one with the values separated by commas. It's now available on the latest version of stash_dio ^3.0.2

**Comment by tjarvstrand** [comment]
Great! Thanks!
